keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaTaskLauncher.kt

```diff
@@ -102,12 +108,28 @@ class OrcaTaskLauncher(
       )
       .let {
         log.info("Started task {} to upsert {}", it.ref, subject)
+
+        if (type == RESOURCE && artifactVersionUnderDeployment != null) {
```

Comment:
This now handles the publishing of the event from a central place, independent of resource type.